openlmis-stock/src/main/java/org/smartregister/stock/openlmis/presenter/StockDetailsPresenter.java

```diff
@@ -154,35 +154,39 @@ private boolean processStockReceived(JSONObject jsonString, String provider) thr
         String date = JsonFormUtils.getFieldValue(stepFields, "Date_Stock_Received");
         String facility = JsonFormUtils.getFieldValue(stepFields, "Receive_Stock_From");
         String reason = JsonFormUtils.getFieldValue(stepFields, "Receive_Stock_Reason");
+        int steps =  Integer.parseInt((String)jsonString.get("count"));
         if (reason.equalsIgnoreCase(OTHER)) {
             reason = JsonFormUtils.getFieldValue(stepFields, "Receive_Stock_Reason_Other");
         }
 
-        return processStock(STEP2, jsonString, provider, date, facility, reason, received);
-
-
+        if (steps == 1) {
+            String status = JsonFormUtils.getFieldValue(stepFields, "Status");
+            int quantity = Integer.parseInt(JsonFormUtils.getFieldValue(stepFields, "Vials_Received"));
+            return processStockWithoutLots(jsonString, provider, date, facility, reason, received, quantity);
+        }
+        return processStockWithLots(STEP2, jsonString, provider, date, facility, reason, received);
     }
 
     private boolean processStockAdjusted(JSONObject jsonString, String provider) throws JSONException {
 
-        return processStock(STEP1, jsonString, provider, simpleDateFormat.format(new Date()),
+        return processStockWithLots(STEP1, jsonString, provider, simpleDateFormat.format(new Date()),
                 null, null, loss_adjustment);
 
     }
 
-    private String extractValue(JSONArray stepFields, String key) throws JSONException {
+    private String extractValue(JSONArray stepFields) throws JSONException {
         for (int i = 0; i < stepFields.length(); i++) {
             JSONObject jsonObject = getJSONObject(stepFields, i);
             String keyValue = jsonObject.getString(KEY);
-            if (STOCK_LOTS.equals(keyValue)) {
-                return jsonObject.optString(key);
+            if (STOCK_LOTS.equals(keyValue) || STOCK_STATUS.equals(keyValue)) {
+                return jsonObject.getString(TRADE_ITEM_ID);
             }
```

Comment:
Return the second param so that its possible to return different keys

Reply:
Done!